controllers/apidefinition_controller.go

```diff
@@ -187,32 +197,44 @@ func (r *ApiDefinitionReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 	return ctrl.Result{}, nil
 }
 
-func ignoreIngressTemplatePredicate() predicate.Predicate {
-	labelFilter := func(labels map[string]string) bool {
-		if isIngressTemplate, ok := labels["template"]; ok {
-			return isIngressTemplate != "true"
-		}
-		return true
+// This triggers an update to all ingress resources that have template
+// annotation matching a.Name.
+//
+// We return nil when a is being deleted and do nothing.
+func (r *ApiDefinitionReconciler) syncTemplate(ctx context.Context, ns string, a *tykv1alpha1.ApiDefinition) error {
+	if !a.DeletionTimestamp.IsZero() {
+		return nil
 	}
-
-	return predicate.Funcs{
-		CreateFunc: func(e event.CreateEvent) bool {
-			return labelFilter(e.Meta.GetLabels())
-		},
-		UpdateFunc: func(e event.UpdateEvent) bool {
-			return labelFilter(e.MetaNew.GetLabels())
-		},
-		DeleteFunc: func(e event.DeleteEvent) bool {
-			// Evaluates to false if the object has been confirmed deleted.
-			return labelFilter(e.Meta.GetLabels())
-		},
+	ls := v1beta1.IngressList{}
+	err := r.List(ctx, &ls,
+		client.InNamespace(ns),
+	)
+	if err != nil {
+		return client.IgnoreNotFound(err)
+	}
+	for _, v := range ls.Items {
+		if v.GetAnnotations()[ingressTemplateAnnotationKey] == a.Name {
```

Comment:
This breaks the API. It is not a label, it's now an annotation. And the `ingressTemplateAnnotationKey` is `tyk.io/template: tyk`, not `template: tyk`